protocols/raft/src/main/java/io/atomix/protocols/raft/storage/snapshot/Snapshot.java

```diff
@@ -71,20 +71,22 @@
    */
   SnapshotChunkReader newChunkReader();
 
-  /**
-   * Completes writing the snapshot to persist it and make it available for reads.
-   *
-   * @return The completed snapshot.
-   */
-  Snapshot complete();
-
   /** Closes the snapshot. */
   @Override
   void close();
 
   /** Deletes the snapshot. */
   void delete();
 
+  /**
+   * Completes writing the snapshot to persist it and make it available for reads.
+   *
+   * @return The completed snapshot.
+   * @deprecated used by the old implementation, superseded {@link PendingSnapshot#commit()}
+   */
+  @Deprecated
```

Comment:
why ? :sweat_smile:

Reply:
I prefer if we have the distinction of complete/incomplete in the types themselves, i.e. `Snapshot` == complete, and `PendingSnapshot` == incomplete. I kept it since the old implementation uses it and I don't want to refactor, but our implementation shouldn't implement it